documentation/triggers.txt

```diff
@@ -1,59 +1,88 @@
 Triggers
 
 AfterSpellCastedTrigger
+Triggers immediately after a spell is cast.
 
 ArmorGainedTrigger
+Triggers whenever target player gains Armor.
 
 BoardChangedTrigger
+Triggers whenever there is a change in the minions/weapons in play. Used for things like checking if Southsea Deckhand should have charge
 
 CardDrawnTrigger
+Triggers whenever target player draws a card.
 
 CardPlayedTrigger
+Triggers whenever target player plays a card.
+
+CardReceivedTrigger
+Triggers whenever target player receives a copy of target card.
 
 DamageCausedTrigger
+Triggers whenever target causes damage.
 
 DamageReceivedTrigger
+Triggers whenever target receives damage.
 
 DiscardTrigger
-
-EnemyAttacksTrigger
+Triggers whenever target player discards a card.
 
 EnrageChangedTrigger
+Triggers whenever the internal ENRAGED attribute of a target is changed.
 
 FatalDamageTrigger
+Triggers whenever target takes fatal damage.
 
 GameStateChangedTrigger
+Triggers whenever anything in the game changes, including things like drawing cards or using hero powers?

```

Comment:
Game States are usually when the state of the game has changed, such as the game starting or ending. I believe this is a higher level one.

On the plus side, you can exchange this Trigger for a different one called "GameStartTrigger" which applies at the start of a game.